app/src/main/java/com/example/sharewhatyoucanproject/donor/DonorActivity.kt

```diff
@@ -0,0 +1,155 @@
+import com.google.firebase.storage.StorageReference
+
+class DonorActivity : AppCompatActivity() {
+
+    lateinit var donorViewModel: DonorViewModel
+    private lateinit var binding: ActivityDonorBinding
+    lateinit var db: FirebaseFirestore
+    lateinit var storage: FirebaseStorage
+    lateinit var storageReference: StorageReference
+    lateinit var locationRequest: LocationRequest
+    lateinit var arrayAdapter: ArrayAdapter<String>
+    lateinit var circularProgressIndicator: com.google.android.material.progressindicator.CircularProgressIndicator
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityDonorBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        circularProgressIndicator = findViewById(R.id.progress_circular)
+        locationRequest = LocationRequest.create()
+        supportActionBar?.hide()
+        storage = FirebaseStorage.getInstance()
+        storageReference = storage.reference
+        db = FirebaseFirestore.getInstance()
+
+        donorViewModel = ViewModelProvider(
+            this,
+            DonorViewModelFactory(applicationContext, db, storageReference),
+        )[DonorViewModel::class.java]
+
+        donorViewModel.currentLocation.observe(this) {
+            it?.let {
+                donorViewModel.uploadImage()
+            }
+        }
+        donorViewModel.imageUrl.observe(this) {
+            it?.let { donorViewModel.saveData() }
+        }
+        donorViewModel.saveData.observe(this) {
+            it?.let {
+                if (it == "success") {
+                    applicationContext.showToast("Post Added")
+                    finish()
+                } else {
+                    circularProgressIndicator.visibility = View.GONE
+                    applicationContext.showToast(it)
+                }
+            }
+        }
+
+        binding.foodimg.setOnClickListener {
+            val intent = Intent()
+            intent.type = "image/*"
+            intent.action = Intent.ACTION_GET_CONTENT
+            startActivityForResult(Intent.createChooser(intent, "Select Picture"), 0)
+        }
+
+        arrayAdapter = ArrayAdapter<String>(
+            applicationContext,
+            android.R.layout.simple_spinner_dropdown_item,
+            foodTypes,
+        )
+        binding.spinner.adapter = arrayAdapter
+        binding.spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(
+                parent: AdapterView<*>?,
+                view: View,
+                position: Int,
+                id: Long,
+            ) {
+                foodTypes[position].also {
+                    donorViewModel.foodType = it
+                    applicationContext.showToast(it)
+                }
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>?) {}
```

Comment:
a nice thing about Kotlin is that if you do not need to use the callback you set it equal to `Unit`:
`override fun onNothingSelected(parent: AdapterView<*>?) = Unit`, however I think you may need it here to clear the food type in the view model since nothing is selected.